test/parallel/test-http2-compat-serverresponse-end-after-statuses-withou-body.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const h2 = require('http2');
+
+// This test case ensures that calling of res.end after sending
+// 204, 205 and 304 HTTP statuses will not cause an error
+// See issue: https://github.com/nodejs/node/issues/21740
+
+const {
+  HTTP_STATUS_NO_CONTENT,
+  HTTP_STATUS_RESET_CONTENT,
+  HTTP_STATUS_NOT_MODIFIED
+} = h2.constants;
+
+const STATUS_WITHOUT_BODY = [
+  HTTP_STATUS_NO_CONTENT,
+  HTTP_STATUS_RESET_CONTENT,
+  HTTP_STATUS_NOT_MODIFIED,
+];
+const STATUS_CODES_COUNT = STATUS_WITHOUT_BODY.length;
```

Comment:
Nit: would you mind changing the `STATUS_WITHOUT_BODY` name to `statusWithoutBody`? Since it is an array that is actually manipulated it is not really a constant. And constants in Node.js are normally written as: e.g. `kStatusCodesCount`.

Reply:
Done.